diff --git a/plugin/x/src/server/builder/ssl_context_builder.h b/plugin/x/src/server/builder/ssl_context_builder.h
index 09b903db..4f5dbe80 100644
--- a/plugin/x/src/server/builder/ssl_context_builder.h
+++ b/plugin/x/src/server/builder/ssl_context_builder.h
@@ -28,6 +28,7 @@

 #include <memory>
 #include <string>
+#include <cstdint>

 #include "plugin/x/src/interface/ssl_context.h"
 #include "plugin/x/src/variables/ssl_config.h"
