diff --git a/extra/zlib/zlib-1.3.2/gzlib.c b/extra/zlib/zlib-1.3.2/gzlib.c
index 7a37a96c..1769fe7e 100644
--- a/extra/zlib/zlib-1.3.2/gzlib.c
+++ b/extra/zlib/zlib-1.3.2/gzlib.c
@@ -3,6 +3,7 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <unistd.h>
 #include "gzguts.h"
 
 #if defined(__DJGPP__)
diff --git a/extra/zlib/zlib-1.3.2/gzread.c b/extra/zlib/zlib-1.3.2/gzread.c
index 8b3e369d..cb29f24c 100644
--- a/extra/zlib/zlib-1.3.2/gzread.c
+++ b/extra/zlib/zlib-1.3.2/gzread.c
@@ -3,6 +3,8 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <fcntl.h>
+#include <unistd.h>
 #include "gzguts.h"
 
 /* Use read() to load a buffer -- return -1 on error, otherwise 0.  Read from
diff --git a/extra/zlib/zlib-1.3.2/gzwrite.c b/extra/zlib/zlib-1.3.2/gzwrite.c
index 13a3700a..2c5363d1 100644
--- a/extra/zlib/zlib-1.3.2/gzwrite.c
+++ b/extra/zlib/zlib-1.3.2/gzwrite.c
@@ -3,6 +3,8 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <fcntl.h>
+#include <unistd.h>
 #include "gzguts.h"
 
 /* Initialize state for writing a gzip file.  Mark initialization by setting
