--- a/libmysql/dns_srv.cc
+++ b/libmysql/dns_srv.cc
@@ -37,6 +37,11 @@
 #include <netdb.h>
 #include <resolv.h>
 
+/* we don't have anything else but the non-tread-safe variants */
+#define res_ninit(X)              (void)X
+#define res_nclose(X)             (void)X
+#define res_nsearch(X,D,I,S,B,L)  res_search(D,I,S,B,L)
+
 // POSIX version
 
 static bool get_dns_srv(Dns_srv_data &data, const char *dnsname, int &error) {

